snapcraft/tests/test_yaml.py

```diff
@@ -449,6 +449,79 @@ def test_tab_in_yaml(self, mock_loadPlugin):
             'on line 2 of snapcraft.yaml')
 
     @unittest.mock.patch('snapcraft.internal.yaml.Config.load_plugin')
+    def test_yaml_valid_epochs(self, mock_loadPlugin):
+        valid_epochs = [
+            0,
+            '1*',
+            1,
+            '400*',
+            1234,
```

Comment:
You mean the string "00001"? That's rejected, to avoid having to validate the number separately from the asterisk in the string. I've updated the tests quite a bit, adding both a test for 01 and "01".